test/widgets/compose_box_test.dart

```diff
@@ -370,11 +420,58 @@ void main() {
       });
     });
 
-    testWidgets('to TopicNarrow', (tester) async {
-      await prepare(tester,
-        narrow: TopicNarrow(channel.streamId, TopicName('topic')));
-      checkComposeBoxHintTexts(tester,
-        contentHintText: 'Message #${channel.name} > topic');
+    group('to ChannelNarrow, mandatory topics', () {
+      final narrow = ChannelNarrow(channel.streamId);
+
+      testWidgets('with empty topic', (tester) async {
+        await prepare(tester, narrow: narrow, mandatoryTopics: true);
+        checkComposeBoxHintTexts(tester,
+          topicHintText: 'Topic',
+          contentHintText: 'Message #${channel.name}');
+      });
+
+      testWidgets('with non-empty but vacuous topic', (tester) async {
+        await prepare(tester, narrow: narrow, mandatoryTopics: true);
+        await enterTopic(tester, narrow: narrow,
+          topic: eg.defaultRealmEmptyTopicDisplayName);
+        await tester.pump();
+        checkComposeBoxHintTexts(tester,
+          topicHintText: 'Topic',
+          contentHintText: 'Message #${channel.name}');
+      });
+
+      testWidgets('legacy: with empty topic', (tester) async {
```

Comment:
nit: the ordering of this among the other test cases doesn't seem logical: it's separating these two non-legacy test cases from each other, but without getting the benefit of being right next to the non-legacy test case that directly corresponds to it.

Either after all three cases, or right after the non-legacy "with empty topic", would be a logical home.